stub Steam::inputActionEventCallback
stub setDualSenseTriggerEffect
remove deprecated/incompatible namespaces ISteamHTMLSurface
InitiateGameConnection_DEPRECATED => InitiateGameConnection
fix/remove dualsense/SCE namespace

Index: godotsteam/godotsteam.cpp
--- godotsteam/godotsteam.cpp.orig
+++ godotsteam/godotsteam.cpp
@@ -1845,7 +1845,7 @@ void Steam::keyChar(uint32 unicode_char, BitField<HTML
 		if (this_handle == 0) {
 			this_handle = browser_handle;
 		}
-		SteamHTMLSurface()->KeyChar(this_handle, unicode_char, (ISteamHTMLSurface::EHTMLKeyModifiers)(int64_t)key_modifiers);
+		SteamHTMLSurface()->KeyChar(this_handle, unicode_char, (EHTMLKeyModifiers)(int64_t)key_modifiers);
 	}
 }
 
@@ -1856,7 +1856,7 @@ void Steam::keyDown(uint32 native_key_code, BitField<H
 		if (this_handle == 0) {
 			this_handle = browser_handle;
 		}
-		SteamHTMLSurface()->KeyDown(this_handle, native_key_code, (ISteamHTMLSurface::EHTMLKeyModifiers)(int64_t)key_modifiers);
+		SteamHTMLSurface()->KeyDown(this_handle, native_key_code, (EHTMLKeyModifiers)(int64_t)key_modifiers);
 	}
 }
 
@@ -1867,7 +1867,7 @@ void Steam::keyUp(uint32 native_key_code, BitField<HTM
 		if (this_handle == 0) {
 			this_handle = browser_handle;
 		}
-		SteamHTMLSurface()->KeyUp(this_handle, native_key_code, (ISteamHTMLSurface::EHTMLKeyModifiers)(int64_t)key_modifiers);
+		SteamHTMLSurface()->KeyUp(this_handle, native_key_code, (EHTMLKeyModifiers)(int64_t)key_modifiers);
 	}
 }
 
@@ -1889,7 +1889,7 @@ void Steam::mouseDoubleClick(HTMLMouseButton mouse_but
 		if (this_handle == 0) {
 			this_handle = browser_handle;
 		}
-		SteamHTMLSurface()->MouseDoubleClick(this_handle, (ISteamHTMLSurface::EHTMLMouseButton)mouse_button);
+		SteamHTMLSurface()->MouseDoubleClick(this_handle, (EHTMLMouseButton)mouse_button);
 	}
 }
 
@@ -1900,7 +1900,7 @@ void Steam::mouseDown(HTMLMouseButton mouse_button, ui
 		if (this_handle == 0) {
 			this_handle = browser_handle;
 		}
-		SteamHTMLSurface()->MouseDown(this_handle, (ISteamHTMLSurface::EHTMLMouseButton)mouse_button);
+		SteamHTMLSurface()->MouseDown(this_handle, (EHTMLMouseButton)mouse_button);
 	}
 }
 
@@ -1922,7 +1922,7 @@ void Steam::mouseUp(HTMLMouseButton mouse_button, uint
 		if (this_handle == 0) {
 			this_handle = browser_handle;
 		}
-		SteamHTMLSurface()->MouseUp(this_handle, (ISteamHTMLSurface::EHTMLMouseButton)mouse_button);
+		SteamHTMLSurface()->MouseUp(this_handle, (EHTMLMouseButton)mouse_button);
 	}
 }
 
@@ -2624,7 +2624,9 @@ bool Steam::setInputActionManifestFilePath(const Strin
 }
 
 // Set the trigger effect for a DualSense controller
-void Steam::setDualSenseTriggerEffect(uint64_t input_handle, int parameter_index, int trigger_mask, SCEPadTriggerEffectMode effect_mode, int position, int amplitude, int frequency) {
+void Steam::setDualSenseTriggerEffect(uint64_t input_handle, int parameter_index, int trigger_mask, int effect_mode, int position, int amplitude, int frequency) {
+	return;
+	/*
 	if (SteamInput() != NULL) {
 		ScePadTriggerEffectParam these_parameters;
 		memset(&these_parameters, 0, sizeof(these_parameters));
@@ -2649,6 +2651,7 @@ void Steam::setDualSenseTriggerEffect(uint64_t input_h
 		these_parameters.command[parameter_index].commandData.vibrationParam.frequency = frequency;
 		SteamInput()->SetDualSenseTriggerEffect((InputHandle_t)input_handle, &these_parameters);
 	}
+	*/
 }
 
 // Waits on an IPC event from Steam sent when there is new data to be fetched from the data drop. Returns true when data was recievied before the timeout expires. Useful for games with a dedicated input thread.
@@ -7758,7 +7761,7 @@ Dictionary Steam::initiateGameConnection(uint64_t serv
 		int authSize = 2048;
 		auth.resize(authSize);
 		CSteamID server = (uint64)server_id;
-		if (SteamUser()->InitiateGameConnection_DEPRECATED(&auth, authSize, server, server_ip, server_port, secure) > 0) {
+		if (SteamUser()->InitiateGameConnection(&auth, authSize, server, server_ip, server_port, secure) > 0) {
 			connection["auth_blob"] = auth;
 			connection["server_id"] = server_id;
 			connection["server_ip"] = server_ip;
@@ -7857,7 +7860,7 @@ void Steam::stopVoiceRecording() {
 // Notify the game server that we are disconnecting. NOTE: This is part of the old user authentication API and should not be mixed with the new API.
 void Steam::terminateGameConnection(uint32 server_ip, uint16 server_port) {
 	if (SteamUser() != NULL) {
-		SteamUser()->TerminateGameConnection_DEPRECATED(server_ip, server_port);
+		SteamUser()->TerminateGameConnection(server_ip, server_port);
 	}
 }
 
@@ -9309,6 +9312,8 @@ void Steam::http_request_headers_received(HTTPRequestH
 //
 // Purpose: when callbacks are enabled this fires each time a controller action state changes
 void Steam::inputActionEventCallback(SteamInputActionEvent_t *call_data) {
+	return;
+	/*
 	uint64_t input_handle = call_data->controllerHandle;
 	int event_type = call_data->eEventType;
 	uint64_t analog_handle = call_data->analogAction.actionHandle;
@@ -9330,6 +9335,7 @@ void Steam::inputActionEventCallback(SteamInputActionE
 	action_information["digital_action_handle"] = digital_handle;
 	action_information["digital_action_data"] = digital_action;
 	emit_signal("input_action_event", input_handle, event_type, action_information);
+	*/
 }
 
 // Purpose: called when a new controller has been connected, will fire once per controller if multiple new controllers connect in the same frame
@@ -13392,13 +13398,13 @@ void Steam::_bind_methods() {
 	BIND_ENUM_CONSTANT(RESULT_PHONE_NUMBER_IS_VOIP);
 
 	// SCEPadTriggerEffectMode Enums
-	BIND_ENUM_CONSTANT(PAD_TRIGGER_EFFECT_MODE_OFF);
-	BIND_ENUM_CONSTANT(PAD_TRIGGER_EFFECT_MODE_FEEDBACK);
-	BIND_ENUM_CONSTANT(PAD_TRIGGER_EFFECT_MODE_WEAPON);
-	BIND_ENUM_CONSTANT(PAD_TRIGGER_EFFECT_MODE_VIBRATION);
-	BIND_ENUM_CONSTANT(PAD_TRIGGER_EFFECT_MODE_MULTIPLE_POSITION_FEEDBACK);
-	BIND_ENUM_CONSTANT(PAD_TRIGGER_EFFECT_MODE_SLOPE_FEEDBACK);
-	BIND_ENUM_CONSTANT(PAD_TRIGGER_EFFECT_MODE_MULTIPLE_POSITION_VIBRATION);
+	//BIND_ENUM_CONSTANT(PAD_TRIGGER_EFFECT_MODE_OFF);
+	//BIND_ENUM_CONSTANT(PAD_TRIGGER_EFFECT_MODE_FEEDBACK);
+	//BIND_ENUM_CONSTANT(PAD_TRIGGER_EFFECT_MODE_WEAPON);
+	//BIND_ENUM_CONSTANT(PAD_TRIGGER_EFFECT_MODE_VIBRATION);
+	//BIND_ENUM_CONSTANT(PAD_TRIGGER_EFFECT_MODE_MULTIPLE_POSITION_FEEDBACK);
+	//BIND_ENUM_CONSTANT(PAD_TRIGGER_EFFECT_MODE_SLOPE_FEEDBACK);
+	//BIND_ENUM_CONSTANT(PAD_TRIGGER_EFFECT_MODE_MULTIPLE_POSITION_VIBRATION);
 
 	// SocketConnectionType Enums
 	BIND_ENUM_CONSTANT(NET_SOCKET_CONNECTION_TYPE_NOT_CONNECTED);
